### DISCUSSION TITLE
Clarify server.properties Strategy and Implement Temporary Whitelist Control via RCON

### DISCUSSION THREAD

**Discussion by AliceCode987** [itzg/docker-minecraft-server#3009] [Q&A] [answered]
Could you please clarify if the strategy for the `server.properties` file is merge or diff? I'm planning to manage the properties file entirely through Docker, but I've encountered some issues.

Specifically, regarding the whitelist: if it's not enabled, there can be issues with script kiddies; if it is enabled, it needs to be configured through environment variables. However, there isn't a way to automatically overwrite default values during Docker creation—other settings must be modified manually, similar to a diff effect. This would mean I wouldn't need to forcefully toggle the whitelist settings.

Additionally, the whitelist requires players to have logged into the server to get their UUID, or they must be added while online using the `whitelist add` command. If I've already joined the server and enabled both the whitelist and enforced whitelist, is it possible to temporarily disable the whitelist via RCON while the server is running to let friends join, then re-enable it afterward? I want to avoid constantly modifying the configuration file and instead have all settings predefined and always enabled.

**Comment by AliceCode987** [comment]
To be more precise, I plan to write certain settings (e.g., world height, MOTD) that are relatively fixed in the Docker Compose environment variables (ENV). This way, I can put the remaining options in the `server.properties` file. Docker will automatically diff these and overwrite the specified settings in `server.properties` with those from the ENV, similar to a partial override. This ensures that the settings provided by Docker's ENV are always consistent, allowing for large-scale deployment, while I can modify all other options in `server.properties` according to the specific requirements of each instance.

**Reply by itzg** [reply, marked as answer]
If I'm understanding your description correctly, it already does what you are wanting

https://docker-minecraft-server.readthedocs.io/en/latest/configuration/server-properties/

Also, whitelist (and ops) can also accept a **username**. Perhaps set the env var `EXISTING_WHITELIST_FILE` to "SYNCHRONIZE" 

If you still have concerns after trying it, please point out which specific variables are not working.

**Comment by AliceCode987** [comment]
it works!
